--- eni.py  2018-04-04 00:52:13.000000000 +0200
+++ eni.py.new  2018-05-29 13:35:51.587281111 +0200
@@ -451,6 +451,12 @@

             if iface.get('name') == "lo":
                 continue
+
+            if nameservers:
+                iface['subnets'][0]["dns_nameservers"] = (" ".join(nameservers))
+            if searchdomains:
+                iface['subnets'][0]["dns_search"] = (" ".join(searchdomains))
+
             sections.extend(
                 self._render_iface(iface, render_hwaddress=render_hwaddress))

